core/tauri/src/window.rs

```diff
@@ -1320,8 +1337,13 @@ impl<R: Runtime> Window<R> {
           payload.inner,
         );
         let resolver = InvokeResolver::new(self, payload.callback, payload.error);
-
         let invoke = Invoke { message, resolver };
+
+        if !is_local && scope.is_none() {
+          invoke.resolver.reject("Scope not defined");
+          return Ok(());
+        }
+
         if let Some(module) = &payload.tauri_module {
```

Comment:
Hummm I don't like that. Currently there is no check for modules like `fs`. I think we should do something similar to plugin where we can whitelist each module.

Reply:
I thought about adding an allowlist configuration object for each external access scope, but maybe the root allowlist can be used for that instead? Any reason a Tauri app should have multiple allowlists, one for each URL?

Also this can be confusing since you'd need to enable the allowlist on the root object to add it to the binary, and also allow it in the external access scope.

Reply:
Yes this is important since it relates to security. Let me give you our usecase.

We have a first in-app window that manages the connections of the user. That window has access to the filesystem to store the configuration.

The user the clicks on a connection which opens a second window with the external URL. We really do not want this window accessing the filesystem directly. We just want it to access certain carefully crafted commands that are safe to call.

This is why I originally only allowed commands and even then I wanted to build a way to limit which commands could be called 

Though I agree there could be a default to allow it to take the same values as the general allow list for modules.

Reply:
Yeah I think we should add an option to block all Tauri APIs. That's easy and won't bloat the codebase with runtime allowlist checks.

Reply:
Pushed a change to address this.